src/pages/ExcelPage.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * ExcelPage.tsx
+ * Example of uploading an excel spreadsheet into DB and displaying the data
+
+
+*/
+import { useState, useEffect } from 'react';
+import { v4 as uuid } from 'uuid';
+
+// material-ui
+import { Box, Typography } from '@mui/material';
+import StaffTable from '../sections/staff/StaffTable';
+import MainCard from '../components/MainCard';
+import { Staff, staffService } from '../sections/staff/staffService';
+
+// sheetJS
+import { read, utils } from "xlsx";
+
+
+const ExcelPage = () => {
+
+    const [uploadStatus, setUploadStatus] = useState('');
+
+    const [staff, setStaff] = useState<Staff[]>([]);
+
+    useEffect(() => {
+        staffService.getStaff()
+            .then(res => setStaff(res))
+    }, [])
+
+    const handleUpload = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        if (!event.target.files) return;
+        const file = event.target.files[0];
+        // get parsed array from the uploaded excel file
+        const newStaffData = await handleParse(file);
+        // upload parsed data to supabase
+        if (newStaffData) {
+            staffService.postStaff(newStaffData)
+                .then(() => {
+                    // append data to state, so changes are reflected in table
+                    setStaff([...staff, ...newStaffData]);
+                    setUploadStatus('Success! Uploaded data to staff table.')
+                })
+                .catch((error) => setUploadStatus(error.toString()));
+        }
+    }
+
+    // referenced example at https://docs.sheetjs.com/docs/demos/frontend/react/
+    const handleParse = async (file: File) => {
```

Comment:
Could we push this function into a service class?  Would make it easier to test.